### PR TITLE
Ask to remove docker yml files during reinit

### PR DESCRIPTION
**Pull request by rpsu**
Allow re-init with a clean state  - add option to remove docker-composer.yml and docker-sync.yml files. 

This allows end users to get the latest goodies even if they concern docker containers configuration (such as #166 )

### CODE REVIEW

File: docker/scripts/ld.command.init.sh [hunk 1]
```diff
@@ -11,9 +11,22 @@ function ld_command_init_exec() {
...
         echo -e "${Yellow}Are you really sure you want to re-initialize the project? ${Color_Off}"
         read -p "[yes/NO] " ANSWER
+        # Lowercase.
+        ANSWER="$(echo ${ANSWER} | tr [A-Z] [a-z])"
         case "$ANSWER" in
             'y'|'yes')
-                echo -e "${Green}Sure, thanks for the confirmation.${Color_Off}"
+                if [ -e "$DOCKERSYNC_FILE" ] || [ -e "$DOCKER_COMPOSE_FILE" ]; then
+                    echo -e "${BYellow}WARNING: ${Yellow}There is docker-compose and/or docker-sync configuration (.yml) files in project root.${Color_Off}"
+                    echo -e "${Yellow}If you continue and rename your volumes ${BYellow}you may loose data (database).${Yellow}.${Color_Off}"
```
Comment (kekkis) [inline docker/scripts/ld.command.init.sh]:
```suggestion
                    echo -e "${Yellow}If you continue and rename your volumes ${BYellow}you may lose data (database).${Color_Off}"
```
One typo, one c-p error?

File: docker/scripts/ld.command.init.sh [hunk 2]
```diff
@@ -11,9 +11,22 @@ function ld_command_init_exec() {
...
+                    echo -e "${Yellow}It is highly recommended to backup your database before continuing:${Color_Off}"
+                    echo -e "${Yellow}./ld db-dump${Color_Off}"
+                    read -p "Remove these files? [Y/n/cancel] " CHOICE
+                    CHOICE="$(echo ${CHOICE} | tr [A-Z] [a-z])"
+                    case "$CHOICE" in
+                        y|'yes'|'') rm -f $DOCKERSYNC_FILE $DOCKER_COMPOSE_FILE 6& echo "Removed." ;;
+                        c|'cancel') echo "Cancelled." && exit 1;;
```
Comment (kekkis) [inline docker/scripts/ld.command.init.sh]: How about `n|'no'`?

Comment (kekkis) [inline docker/scripts/ld.command.init.sh]:
```suggestion
                        c|'cancel') echo "Cancelled reinitialization of docker-compose/docker-sync config." && exit 1;;
```
More explicit.

File: docker/scripts/ld.command.init.sh [hunk 3]
```diff
@@ -62,24 +77,26 @@ function ld_command_init_exec() {
...
-          VALID=1
-      else
-          echo -e "${Red}ERROR: Domain name can contain only alphabetic characters (a-z), numbers (0-9), hyphens (-), underscoreds (_) and dots (.).${Color_Off}"
-          echo -e "${Red}ERROR: Also the domain name must not start or end with underscore, hyphen or dot.${Color_Off}"
-          sleep 2
-          echo
-      fi
+        echo
+        echo  -e "${BBlack}== Local development base domain == ${Color_Off}"
+        echo -e "Do not add protocol nor www -part but just the domain name. It is recommended to use domain ending with ${BBlack}.local${Black}.${Color_Off}"
```
Comment (kekkis) [inline docker/scripts/ld.command.init.sh]:
```suggestion
        echo -e "Do not add protocol nor www part but just the domain name. It is recommended to use domain ending with ${BBlack}.local${Black}.${Color_Off}"
```
One less hyphen

File: docker/scripts/ld.command.init.sh [hunk 4]
```diff
@@ -62,24 +77,26 @@ function ld_command_init_exec() {
...
+        # Lowercase.
+        ANSWER="$(echo ${ANSWER} | tr [A-Z] [a-z])"
+        TEST=$(echo $ANSWER | egrep -e '^(([a-zA-Z0-9])([a-zA-Z0-9\.]*))?([a-zA-Z0-9])$')
+        if [ -z "$ANSWER" ]; then
+            VALID=1
+        elif [ "${#TEST}" -gt 0 ]; then
+            LOCAL_DOMAIN=$ANSWER
+            VALID=1
+        else
+            echo -e "${Red}ERROR: Domain name can contain only alphabetic characters (a-z), numbers (0-9), hyphens (-), underscoreds (_) and dots (.).${Color_Off}"
```
Comment (kekkis) [inline docker/scripts/ld.command.init.sh]:
```suggestion
            echo -e "${Red}ERROR: Domain name can contain only alphabetic characters (a-z), numbers (0-9), hyphens (-), underscores (_) and dots (.).${Color_Off}"
```
typo

File: docker/scripts/ld.command.init.sh [hunk 5]
```diff
@@ -62,24 +77,26 @@ function ld_command_init_exec() {
...
-      fi
+        echo
+        echo  -e "${BBlack}== Local development base domain == ${Color_Off}"
+        echo -e "Do not add protocol nor www -part but just the domain name. It is recommended to use domain ending with ${BBlack}.local${Black}.${Color_Off}"
+        DEFAULT=${PROJECT_NAME}.local
+        LOCAL_DOMAIN=${LOCAL_DOMAIN:-${DEFAULT}}
+        read -p "Domain [$LOCAL_DOMAIN] " ANSWER
+        # Lowercase.
+        ANSWER="$(echo ${ANSWER} | tr [A-Z] [a-z])"
+        TEST=$(echo $ANSWER | egrep -e '^(([a-zA-Z0-9])([a-zA-Z0-9\.]*))?([a-zA-Z0-9])$')
```
Comment (kekkis) [inline docker/scripts/ld.command.init.sh]: This regex does not seem to accept `-` or `_` as the text below would suggest.

File: docker/scripts/ld.command.init.sh (hunk 2 above)
Comment (rpsu) [reply docker/scripts/ld.command.init.sh]: good idea, thanks. 
6e68b29

Comment (rpsu) [reply docker/scripts/ld.command.init.sh]: thanks, very good.
201e61d

File: docker/scripts/ld.command.init.sh (hunk 5 above)
Comment (rpsu) [reply docker/scripts/ld.command.init.sh]: I don't think underscore is really needed for local development domain, so removed that part in the suggestion. Similar change also in project name for the same reason.

Comment (rpsu) [reply docker/scripts/ld.command.init.sh]: bc9246e

Comment (rpsu) [reply docker/scripts/ld.command.init.sh]: bc9246e

**Pull request merged**